pkg/cmd/codespace/ssh.go

```diff
@@ -121,17 +121,19 @@ func (a *App) SSH(ctx context.Context, sshArgs []string, opts sshOptions) (err e
 		args = opts.scpArgs
 	}
 
-	sshContext := ssh.SshContext{}
-	startSshServerOpts := []func(*liveshare.StartSSHServerOptions){}
+	sshContext := ssh.Context{}
+	startSSHOptions := liveshare.StartSSHServerOptions{}
 
 	if shouldGenerateSSHKeys(args, opts) && sshContext.HasKeygen() {
-		keyPair, err := sshContext.GenerateSSHKey("codespaces", ssh.WithNoErrorOnExistingKey())
+		keyPair, err := sshContext.GenerateSSHKey("codespaces", "")
 		if err != nil {
-			return fmt.Errorf("failed to generate ssh keys: %s", err)
+			if _, ok := err.(*ssh.KeyAlreadyExistsError); !ok {
```

Comment:
Errors should be compared using `errors.Is()` or `errors.As()` so that the comparison stays valid even if the KeyAlreadyExistsError happened to be wrapped